### ISSUE TITLE
[CP][stable channel] Ensure start/stop file watching requests are run on the dart thread (Additional Change)

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#46298]
commit(s) to merge: 05e5427800f43d0a9bac6b3a77e8e5f65c84812c

merge instructions: clean merge

What is the issue: 

A cherry pick for the original problem was done in https://github.com/dart-lang/sdk/issues/46249, but another issue was noted during the cherry pick review which is addressed by this change.

numerous user reports of the analysis server stopping responding to IDE requests; this is due to the isolate running the analysis server deadlocking when unsubscribing from file watching events

What is the fix: 

At present start/stop requests are scheduled on RunLoop thread. This results in deadlocks since same RunLoop thread might be busy with blocking writes of file watching events, not giving a chance for Dart to read previously-written events. Reading would unblock writer.

This fix moves start/stop requests to run on Dart thread instead.

Why cherrypick: 
lots of users reports; likely manifesting in a few separate issues
Risk:

The change has rolled into flutter without any issues.

Link to original issue(s):

#45996
/cc @kevmoo @mit-mit @whesse @athomas @vsmenon @franklinyow @a-siva @aam @DanTup

**Comment by franklinyow** [comment]
Description:
Additional fix for analysis server stop responding to IDE requests , see #45996 for detail

Approved in SCRUM
CC: @kf6gpe

**Comment by athomas** [comment]
This doesn't merge cleanly into stable:
```
$ git diff
diff --cc runtime/bin/main_options.h
index 658cea3fd3d,4fab09cc36d..00000000000
--- a/runtime/bin/main_options.h
+++ b/runtime/bin/main_options.h
@@@ -48,7 -48,9 +48,13 @@@ namespace bin 
    V(suppress_core_dump, suppress_core_dump)                                    \
    V(enable_service_port_fallback, enable_service_port_fallback)                \
    V(disable_dart_dev, disable_dart_dev)                                        \
++<<<<<<< HEAD
 +  V(long_ssl_cert_evaluation, long_ssl_cert_evaluation)
++=======
+   V(long_ssl_cert_evaluation, long_ssl_cert_evaluation)                        \
+   V(bypass_trusting_system_roots, bypass_trusting_system_roots)                \
+   V(delayed_filewatch_callback, delayed_filewatch_callback)
++>>>>>>> 05e5427800f ([io/mac] Ensure FSEventsWatcher::Node is deleted synchronously with Callback that uses it.)
  
  // Boolean flags that have a short form.
  #define SHORT_BOOL_OPTIONS_LIST(V)
```

**Comment by athomas** [comment]
Merged to stable in 516eb3b05c252ef3d34255d06700c1f09f7f4502 (2.13.3).
